rcos/Telescope
# 127 add associate command

Issue Number: close #127

Problem Summary:
Add `/associate` slash command to Discord bot.

### What is changed and how it works?
**graphql/rcos/discord_associations: "project/find_project.graphql", "small_group/find_small_group.graphql", src/api/rcos/discord_assoications: "project/project_info.rs", "small_group/small_group_info.rs",
src/discord_bot/commands: "associate.rs"**
#### associate.rs:
- Implemented `/associate` command.
- `/associate` command has 3 subcommand groups: `/associate channel`, `/associate role`, and `/associate category`.
- Each subcommand group has 2 sub commands: `/associate <subcommand group> project` and `/associate <subcommand group> small group`.
- Each command requires user to enter two optional value:
  - The first option is: <channel_name>, <category_name> or <role_name>.
  - The second option is: <project_id> or <small_group_id>, which is corresponding to the project/small group id in the database.
- Each command will associate existing channels/roles with projects/small groups. 
- This command is limited to coordinators, faculty advisors, and sysadmins.

### Tests


Manual test:
1.  Create Discord Application.
2.  Create a new discord server.
3. Create roles "Faculty Advisors" and "Coordinator", add yourself to these two roles.
4. Start the docker image.
5. Invite bots to discord server using the link generated by Telescope.
6. Open Hasura console.
7. Insert one row into small groups table.
8. Insert one row into projects table.

For `/associate category` command:
1. Create a category on discord server.
2. Using command `/associate category smallgroup <category name> <small group id>` to associate newly created category with small group. Discord bot will return success message and added data into small_group_categories table.
3. Create a text/voice channel on discord server.
4. Using command `/associate category smallgroup <category name> <small group id>` to associate newly created category with small group. Discord bot will return error message and indicate that user should select a valid voice/text channel.

For `/associate channel` command:
1. Create a channel on discord server.
2. Using command `/associate channel project <channel name> <project id>` to associate newly created channel with project. Discord bot will return success message and added data into project_channels table.
3. Create a category on discord server.
4. Using command `/associate channel project <channel name> <project id>` to associate newly created category with project. Discord bot will return error message and indicate that user should select a category.

For `/associate role` command:
1. Create two role on discord server.
2. Using command `/associate role project <role name> <project id>` to associate one of newly created role with project. Discord bot will return success message and added data into project_roles table.
3. Using command `/associate role small group <role name> <small group id>` to associate one of newly created role with project. Discord bot will return success message and added data into small_group_roles table.
